### DISCUSSION TITLE
Add headers as part of the request for an external context using version 5.0.0

### DISCUSSION THREAD

**Discussion by matharut** [eclipse-rdf4j/rdf4j#5068] [General]
Using RDF4J 5.0.0 is it possible to send additional headers when the the context is external?  So for example if I have a context points to `https://my.site.com/schema/context/ ` which is included in the string `jsonld` but I want to also pass in some additional headers.  Below is the code I have but not sure how to add the headers

```
        Model model = new LinkedHashModel();
        try (InputStream inputStream = new ByteArrayInputStream(jsonld.getBytes())) {
            Set<String> whitelist = Set.of("https://my.site.com/schema/context/");
            ParserConfig config = new ParserConfig();
            config.set(JSONLDSettings.WHITELIST, whitelist);
            model = Rio.parse(inputStream, "", RDFFormat.JSONLD, config);

        } catch (IllegalArgumentException e) {
           log.error("illegal arg exception", e);
        } catch (Exception e) {
            log.error("Exception parsing model", e);
        }
```

**Comment by hmottestad** [comment]
This should be solved by injecting your own loader, but I can see now that I forgot to expose that for the new JSON-LD parser. We can expose it in RDF4J 5.1.0. 

You can set a fallback content type if the content type is the main issue you are having:    https://github.com/HASMAC-AS/hasmac-json-ld/blob/b6ab827821cc5f915854346fd3b7338c6a7f17e1/src/main/java/no/hasmac/jsonld/loader/DefaultHttpLoader.java#L222

**Reply by matharut** [reply]
I would like to add 'Accept' header with the profile, i.e.  `application/ld+json;profile=http://www.w3.org/ns/json-ld#context`.  It looks like I could set this if I was able to inject DocumentLoaderOptions
When would 5.1.0 be released?

**Reply by hmottestad** [reply]
We don't currently have a timeline for 5.1.0. A rough estimate would be sometime in the autumn. 

It would be very helpful if you could create an issue and contribute a test case and solution :) That always helps speed things along.

**Reply by hmottestad** [reply]
Btw. As a workaround you could download the context yourself and then inject it as an expand context.

**Reply by matharut** [reply]
Thanks, yes I am thinking the same. Makes it a larger document to parse particularly if the context is big and hence impact on performance

**Reply by hmottestad** [reply]
Not sure what you mean. The context is always fully resolved and parsed. There is no lazy loading as far as I can remember.

**Reply by vorburger** [reply]
#5080 has a related discussion.
